arm-operationalinsights/2015-03-20/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1142 @@
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Resource Group name of the Log Analytics Workspace"
+          },
+          {
+            "name": "workspaceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Log Analytics Workspace name that will contain the storageInsightsConfigs resource"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "$ref": "#/definitions/StorageInsightListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "@odata.nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.OperationalInsights/linkTargets": {
+      "get": {
+        "tags": [
+          "Workspaces"
+        ],
+        "operationId": "Workspaces_ListLinkTargets",
+        "description": "Gets a list of accounts with administrator privileges that are not associated with an existing workspace.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Account"
+              }
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/savedSearches/{savedSearchId}": {
+      "delete": {
+        "tags": [
+          "SavedSearch"
+        ],
+        "operationId": "Search_DeleteSavedSearch",
+        "description": "Deletes the specified saved search in a given workspace.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Resource Group name of the Log Analytics Workspace"
+          },
+          {
+            "name": "workspaceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Log Analytics workspace name"
+          },
+          {
+            "name": "savedSearchId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The id of the saved search."
```

Comment:
Are there constraints on the name (ie, `maxLength` and `minLength` and perhaps a regex `pattern` ?)

At least with a `maxLength` the client will ensure it doesn't send insane strings to the service. :D

---
In reply to: [87509333](https://github.com/Azure/azure-rest-api-specs/pull/662#discussion_r87509333) [](ancestors = 87509333)